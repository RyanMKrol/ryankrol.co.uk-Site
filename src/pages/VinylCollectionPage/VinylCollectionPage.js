import React, { Component } from 'react';

import styles from './VinylCollectionPage.module.css';

import { DataCollection, VinylCollectionItem } from '../../components';
import { baseEndpoint } from '../../utils/ProdUtils';

/**
 * Page for rendering sub-section of Ratings content
 */
class VinylCollectionPage extends Component {
  /**
   * @class
   */
  constructor() {
    super();

    this.endpoint = `${baseEndpoint()}/vinyl`;
  }

  /**
   * Render method
   *
   * @returns {string} JSX Content
   */
  render() {
    return (
      <div className={styles.container}>
        <h1 className="light">My Vinyl Collection</h1> <br />
        <p>Literally just a bunch of vinyl I own, idk</p>
        <DataCollection endpoint={this.endpoint} itemTag={VinylCollectionItem} />
      </div>
    );
  }
}

export default VinylCollectionPage;
