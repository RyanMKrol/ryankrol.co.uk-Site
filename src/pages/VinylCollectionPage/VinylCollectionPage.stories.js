import React from 'react';

import VinylCollectionPage from './VinylCollectionPage';

export default {
  title: 'Pages/VinylCollectionPage',
};

export const Default = () => <VinylCollectionPage />;
