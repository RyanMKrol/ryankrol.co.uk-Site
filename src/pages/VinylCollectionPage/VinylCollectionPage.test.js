import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import VinylCollectionPage from './VinylCollectionPage';

it('renders VinylCollectionPage correctly', () => {
  const header = shallow(<VinylCollectionPage />);
  expect(toJson(header)).toMatchSnapshot();
});
