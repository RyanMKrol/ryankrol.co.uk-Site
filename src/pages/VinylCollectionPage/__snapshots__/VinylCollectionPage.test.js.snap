// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders VinylCollectionPage correctly 1`] = `
<div
  className="container"
>
  <h1
    className="light"
  >
    My Vinyl Collection
  </h1>
   
  <br />
  <p>
    Literally just a bunch of vinyl I own, idk
  </p>
  <DataCollection
    endpoint="http://www.ryankrol.co.uk/api/vinyl"
    itemTag={[Function]}
  />
</div>
`;
